test/flows/child_benefit_tax_calculator_flow_test.rb

```diff
@@ -0,0 +1,399 @@
+        assert_next_node :results, for_response: "no"
+      end
+    end
+  end
+
+  context "question: other_allowable_deductions?" do
+    setup do
+      testing_node :other_allowable_deductions?
+      add_responses how_many_children?: "5",
+                    which_tax_year?: "2021",
+                    is_part_year_claim?: "no",
+                    income_details?: "100",
+                    add_allowable_deductions?: "yes",
+                    allowable_deductions?: "100",
+                    add_other_allowable_deductions?: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of results" do
+        assert_next_node :results, for_response: "100"
+      end
+    end
+  end
+
+  context "outcome: results" do
+    setup { testing_node :results }
+
+    should "render no tax charge when the net income is below £51,000" do
+      add_responses how_many_children?: "1",
+                    which_tax_year?: "2021",
+                    is_part_year_claim?: "no",
+                    income_details?: "60000",
+                    add_allowable_deductions?: "yes",
+                    allowable_deductions?: "5000",
+                    add_other_allowable_deductions?: "yes",
+                    other_allowable_deductions?: "10000"
+
+      assert_rendered_outcome text: "There is no tax charge if your income is below £50,099"
+    end
+
+    should "render the tax owed the net income is above £51,000" do
+      add_responses how_many_children?: "2",
+                    which_tax_year?: "2020",
+                    is_part_year_claim?: "yes",
+                    how_many_children_part_year?: "2",
+                    child_benefit_1_start?: "2020-09-07",
+                    add_child_benefit_1_stop?: "yes",
+                    child_benefit_1_stop?: "2020-09-13",
+                    child_benefit_2_start?: "2020-09-07",
+                    add_child_benefit_2_stop?: "yes",
+                    child_benefit_2_stop?: "2020-09-13",
+                    income_details?: "100000",
+                    add_allowable_deductions?: "no"
+
+      # Two weeks of child benefit for 2020 tax year = 21.05 + 13.95
+      assert_rendered_outcome text: "The estimated tax charge to pay is £35.00"
+    end
+
+    should "render estimate text when income is above £51,000 and the tax year is incomplete" do
+      travel_to("2021-09-01") do
+        add_responses how_many_children?: "1",
+                      which_tax_year?: "2021",
+                      is_part_year_claim?: "no",
+                      income_details?: "100000",
+                      add_allowable_deductions?: "no"
+
+        assert_rendered_outcome text: "This is an estimate based on your adjusted net income of £100,000.00"
+      end
+    end
+
+    should "render specific guidance when the tax year is 2012-2013" do
```

Comment:
There doesn't seem to be an outcome test for [this other 2012 if statement](https://github.com/alphagov/smart-answers/blob/main/app/flows/child_benefit_tax_calculator_flow/outcomes/results.erb#L59)

Reply:
Our precedence is to just test conditionals once in outcomes - we get the same coverage either way. Otherwise it'd be pretty arduous in some cases where the same conditional is multiple times in an outcome.

Reply:
👍